examples/04_vcb_tool/ex_vcb_tool/main.rb

```diff
@@ -0,0 +1,208 @@
+        if picked_first_point?
+          @mouse_ip.pick(view, x, y, @picked_first_ip)
+
+          # Print the length of the previewed line to the measurement bar.
+          # Printing a Length object (as opposed to a float) automatically
+          # formats it according to the model units.
+          Sketchup.vcb_value = @mouse_ip.position.distance(@picked_first_ip.position)
+        else
+          @mouse_ip.pick(view, x, y)
+        end
+        view.tooltip = @mouse_ip.tooltip if @mouse_ip.valid?
+        view.invalidate
+      end
+
+      def onLButtonDown(flags, x, y, view)
+        # Track where in screen space mouse is pressed down.
+        @mouse_down = Geom::Point3d.new(x, y)
+
+        if picked_first_point? && create_edge > 0
+          # When the user have picked a start point and then picks another point
+          # we create an edge and try to create new faces from that edge.
+          # Like the native tool we reset the tool if it created new faces.
+          reset_tool
+        else
+          # If no face was created we let the user chain new edges to the last
+          # input point.
+          @picked_first_ip.copy!(@mouse_ip)
+        end
+
+        update_ui
+        view.invalidate
+      end
+
+      def onLButtonUp(flags, x, y, view)
+        if @mouse_down.distance([x, y]) > DRAG_THRESHOLD
+          # Mouse is released far enough away from where it was pressed to be
+          # considered to be dragged there.
+
+          create_edge
+
+          # When drag-drawing, it feels odd to chain line drawing and we
+          # consistently reset the tool instead.
+          # You can try only calling this method if create_edge returns a
+          # positive number, to experience the other behavior.
+          reset_tool
+        end
+      end
+
+      # Here we have hard coded a special ID for the pencil cursor in SketchUp.
+      # Normally you would use `UI.create_cursor(cursor_path, 0, 0)` instead
+      # with your own custom cursor bitmap:
+      #
+      #   CURSOR_PENCIL = UI.create_cursor(cursor_path, 0, 0)
+      CURSOR_PENCIL = 632
+
+      def onSetCursor
+        # Note that `onSetCursor` is called frequently so you should not do much
+        # work here. At most you switch between different cursor representing
+        # the state of the tool.
+        UI.set_cursor(CURSOR_PENCIL)
+      end
+
+      # Listen to measurement bar text input.
+      def onUserText(text, view)
+        distance = text.to_l
+        direction = picked_points[1] - picked_points[0]
+        end_point = picked_points[0].offset(direction, distance)
+        @mouse_ip = Sketchup::InputPoint.new(end_point)
+
+        # If faces are created when creating the edge, reset the tool.
+        # Otherwise keep drawing edges.
+        if create_edge > 0
+          reset_tool
+        else
+          @picked_first_ip.copy!(@mouse_ip)
+        end
+
+        view.invalidate
+      rescue ArgumentError
```

Comment:
I think this really should be wrapped around `distance = text.to_l`.

If there's an error (bug) elsewhere it might also raise an `ArgumentError`. The capture block is really handling error for that one line, so it shouldn't capture errors for anything else. Anything else would be a bug and should not be muted by this `rescue`.

Reply:
Good point